AudioSinkExample/AudioSinks/ExampleSpeechRecognizer.m

```diff
@@ -0,0 +1,142 @@
+//
+//  ExampleSpeechRecognizer.m
+//  AudioSinkExample
+//
+//  Copyright © 2017 Twilio, Inc. All rights reserved.
+//
+
+#import "ExampleSpeechRecognizer.h"
+
+#import <AudioToolbox/AudioToolbox.h>
+
+@interface ExampleSpeechRecognizer()
+
+@property (nonatomic, strong) SFSpeechRecognizer *speechRecognizer;
+@property (nonatomic, strong) SFSpeechAudioBufferRecognitionRequest *speechRequest;
+@property (nonatomic, strong) SFSpeechRecognitionTask *speechTask;
+@property (nonatomic, assign) AudioConverterRef speechConverter;
+@property (nonatomic, assign) int numberOfChannels;
+
+@property (nonatomic, copy) NSString *speechResult;
+@property (nonatomic, weak) TVIAudioTrack *audioTrack;
+
+@end
+
+@implementation ExampleSpeechRecognizer
+
+- (instancetype)initWithAudioTrack:(TVIAudioTrack *)audioTrack
+                        identifier:(NSString *)identifier
+                     resultHandler:(void (^)(SFSpeechRecognitionResult * result, NSError * error))resultHandler {
+    self = [super init];
+
+    if (self != nil) {
+        _speechRecognizer = [[SFSpeechRecognizer alloc] init];
+        _speechRecognizer.defaultTaskHint = SFSpeechRecognitionTaskHintDictation;
+
+        _speechRequest = [[SFSpeechAudioBufferRecognitionRequest alloc] init];
+        _speechRequest.shouldReportPartialResults = YES;
+
+        // The iOS audio device captures in mono.
+        // The mixer produces stereo audio for each remote Participant, even if they send mono audio.
+        _numberOfChannels = [audioTrack isKindOfClass:[TVILocalAudioTrack class]] ? 1 : 2;
+
+        __weak typeof(self) weakSelf = self;
+        _speechTask = [_speechRecognizer recognitionTaskWithRequest:_speechRequest resultHandler:^(SFSpeechRecognitionResult * _Nullable result, NSError * _Nullable error) {
+            __strong typeof(self) strongSelf = weakSelf;
+            if (result) {
+                strongSelf.speechResult = result.bestTranscription.formattedString;
+            } else {
+                // TODO: CE - do we handle errors or let our owner?
+                NSLog(@"Speech recognition error: %@", error);
+            }
```

Comment:
This was one question I had - who should be handling errors and ultimately call `stopRecognizing`. Should it be the ViewController which owns `ExampleSpeechRecognizer`?

Reply:
I agree with you, lets make the `ViewController` the class which responds to the errors.